src/coreclr/jit/morph.cpp

```diff
@@ -12121,6 +12121,63 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
                 }
                 break;
             }
+            case NI_SSE_And:
+            case NI_SSE2_And:
+            case NI_AVX_And:
+            case NI_AVX2_And:
+            {
+                // Transforms ~v1 & v2 to VectorXxx.AndNot(v1, v2)
+                GenTree* op1 = hw->Op(1);
+                GenTree* op2 = hw->Op(2);
+                if (!gtIsActiveCSE_Candidate(hw))
+                {
+                    if (op1->OperIs(GT_HWINTRINSIC))
+                    {
+                        NamedIntrinsic      ni       = NI_Illegal;
+                        GenTreeHWIntrinsic* inner_hw = op1->AsHWIntrinsic();
+                        switch (inner_hw->GetHWIntrinsicId())
+                        {
+                            case NI_SSE_Xor:
```

Comment:
I think it should be:
```c
case Vector128_Xor:
case NI_SSE2_Xor:
case NI_SSE_Xor:
    ni = NI_Vector128_AndNot;
    break;

case Vector256_Xor:
case NI_AVX_AndNot:
case NI_AVX2_AndNot:
    ni = NI_Vector256_AndNot;
    break;
```

Reply:
Also, keep in mind that `NI_VectorXXX_AndNot` and e,g, `NI_SSE2_AndNot` aren't the same - one of them is `x & ~y` the other is `~x & y`

Reply:
`gtNewBinOpNode` handles this difference for you, thankfully. You pass it in as `~x, y` and it will do the right thing for the constructed intrinsic node.